MinecraftClient/Translations.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Text;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace MinecraftClient
+{
+    /// <summary>
+    /// Allows to localize the app in different languages
+    /// </summary>
+    /// <remarks>
+    /// By ORelio (c) 2015-2018 - CDDL 1.0
+    /// </remarks>
+    public static class Translations
+    {
+        private static Dictionary<string, string> translations;
+
+        /// <summary>
+        /// Return a tranlation for the requested text
+        /// </summary>
+        /// <param name="msg_name">text identifier</param>
+        /// <returns>returns translation for this identifier</returns>
+        public static string Get(string msg_name, params object[] args)
+        {
+            if (translations.ContainsKey(msg_name))
+            {
+                if (args.Length > 0)
+                {
+                    return string.Format(translations[msg_name], args);
+                }
+                else return translations[msg_name];
+            }
+                
+
+            return msg_name.ToUpper();
+        }
+
+        /// <summary>
+        /// Initialize translations depending on system language.
+        /// English is the default for all unknown system languages.
+        /// </summary>
+        static Translations()
+        {
+            translations = new Dictionary<string, string>();
+
+            /*
+             * External translation files
+             * These files are loaded from the installation directory as:
+             * Lang/abc.ini, e.g. Lang/eng.ini which is the default language file
+             * Useful for adding new translations of fixing typos without recompiling
+             */
+
+            string systemLanguage = CultureInfo.CurrentCulture.ThreeLetterISOLanguageName;
+            string langDir = AppDomain.CurrentDomain.BaseDirectory + Path.DirectorySeparatorChar + "MCC_Lang" + Path.DirectorySeparatorChar;
+            string langFileSystemLanguage = langDir + systemLanguage + ".ini";
+            string langFileDefault = langDir + "eng.ini";
+
+            string langFileToUse = langFileDefault;
+            if (File.Exists(langFileSystemLanguage))
+                langFileToUse = langFileSystemLanguage;
+
+            // Write the language file for Eng if does not exist
+            if (!File.Exists(langFileDefault))
+                WriteDefaultLang();
+
+            foreach (string lineRaw in File.ReadAllLines(langFileToUse, Encoding.UTF8))
```

Comment:
Tip: I did not think about this when writing `Translations.cs`, but it is likely that MCC will have incomplete translation files because of new messages introduced in new versions. I'd suggest to load `langFileDefault` first, then load `langFileToUse` to overwrite the first translations so English translations will be kept for missing messages.